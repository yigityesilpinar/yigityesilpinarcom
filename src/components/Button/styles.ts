import styled, { css } from 'styled-components'

import LoadingIcon from './LoadingIcon'
import { ButtonProps } from './Button'

export const StyledButton = styled.button<Omit<ButtonProps, 'iconSrc' | 'text'>>`
  display: flex;
  align-items: center;
  position: relative;
  background-color: ${({ theme }) => theme.palette.primary.main};
  color: ${({ theme }) => theme.palette.primary.contrastText};
  border: 1px solid ${({ theme }) => theme.palette.primary.main};
  border-radius: 3em;
  min-height: 3em;
  padding: 0 1.5em;
  cursor: pointer;
  img {
    height: 2em;
    margin-right: 0.5em;
  }
  :hover {
    opacity: 0.7;
  }

  ${({ isLoading }) =>
    isLoading &&
    css`
      opacity: 0.8;
      :hover {
        opacity: 0.8;
      }
      img,
      span {
        opacity: 0;
      }
    `}
`

export const StyledLoadingIcon = styled(LoadingIcon)`
  position: absolute;
  margin: 0 auto;
  left: 0;
  right: 0;
`
