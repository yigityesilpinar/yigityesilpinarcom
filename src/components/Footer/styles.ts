import styled from 'styled-components'

export const FooterContainer = styled.div`
  display: flex;
  align-items: center;
  height: 4.5em;
  background-color: ${({ theme }) => theme.palette.text.primary};
  padding-left: 7.625em;
  padding-right: 3em;
`

export const FooterLinksContainer = styled.div`
  display: flex;
  align-items: center;
  margin-left: auto;
`

export const FooterLink = styled.a`
  img {
    height: 2.5em;
  }
  :not(:first-child){
    margin-left: 1em;
  }
`
