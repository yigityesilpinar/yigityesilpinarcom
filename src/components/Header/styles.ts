import styled from 'styled-components'

export const HeaderContainer = styled.div`
  display: flex;
  align-items: center;
  height: 5em;
  padding-left: 7.625em;
  a {
    display: flex;
    height: 100%;
    align-items: center;
    position: relative;
    color: ${({ theme }) => theme.palette.text.primary};
    padding: 0 1.5em;
    :hover {
      opacity: 0.7;
    }
    &:first-child {
      padding-left: 0;
      &:after {
        content: '';
        position: absolute;
        border-bottom: 1px solid black;
        bottom: 0;
        left: -7.625em;
        width: 0;
        opacity: 0;
        transition: all .6s ease-in-out;
      }
    }
    &:not(:first-child):after {
      content: '';
      position: absolute;
      border-bottom: 1px solid black;
      bottom: 0;
      left: 0;
      right: 0;
      width: 0;
      transition: all .6s ease-in-out;
      opacity: 0;
    }
    &.active:after {
      opacity: 1;
      width: 100%;
    }
    &.active:first-child:after {
      width: calc(100% + 7.625em);
    }
  }
`
