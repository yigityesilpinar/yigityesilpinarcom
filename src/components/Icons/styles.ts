import styled from 'styled-components'

export const Container = styled.div`
  border-radius: 50%;
  display: inline-block;
  margin: 0 4px;
  padding: 12px;
  overflow: hidden;
  vertical-align: middle;
  cursor: pointer;
  height: 24px;
  width: 24px;
  flex: 0 0 auto;
  :hover {
    background-color: rgba(60, 64, 67, 0.08);
    outline: none;
  }
  svg {
    fill: ${(props) => props.theme.palette.secondary.main};
    opacity: 1;
  }
`
