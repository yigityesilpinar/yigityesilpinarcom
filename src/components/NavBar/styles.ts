import styled from 'styled-components'

export const Container = styled.div``

export const LinksContainer = styled.div`
  display: flex;
  flex-direction: column;
  position: absolute;
  box-shadow: 0 1px 2px 0 rgba(60, 64, 67, 0.3),
    0 1px 3px 1px rgba(60, 64, 67, 0.15);
  top: 0;
  left: 0;
  min-width: 300px;
  padding: 20px;
  height: 90vh;
  background: white;
`
