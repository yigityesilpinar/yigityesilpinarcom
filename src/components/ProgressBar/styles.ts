import styled from 'styled-components'

export const Progress = styled.div`
  position: absolute;
  height: 100%;
  z-index: 1;
  border-radius: 0.5em;
`

export const Total = styled.div`
  position: absolute;
  height: 100%;
  width: 100%;
  border-radius: 0.5em;
`

export const ProgressBarContainer = styled.div<{ percantage: number; progressColor: string; totalColor: string }>`
  position: relative;
  height: 0.7em;
  ${Progress} {
    width: ${({ percantage }) => percantage}%;
    background-color: ${({ progressColor }) => progressColor};
  }
  ${Total} {
    background-color: ${({ totalColor }) => totalColor};
  }
`
