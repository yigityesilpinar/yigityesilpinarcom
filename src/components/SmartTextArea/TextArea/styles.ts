import styled, { css } from 'styled-components'
import _TextareaAutosize from 'react-textarea-autosize'

export const Container = styled.div`
  position: relative;
  width: 100%;
`

export const textAreaSpacings = css`
  line-height: 1.5;
  padding: 4px 11px;
`

export const TextareaAutosize = styled(_TextareaAutosize)<{
  $hasError?: boolean
}>`
  font-family: 'Open Sans', sans-serif;
  font-size: inherit;
  border: 1px solid ${({ theme }) => theme.palette.secondary.main};
  border-radius: 2px;
  ${textAreaSpacings}
  width: 100%;
  resize: none;
  ${({ disabled }) => disabled && 'cursor: not-allowed; background: #f5f5f5;'};
  &:hover,
  &:active,
  &:focus {
    border-color: ${({ theme, disabled }) => !disabled && theme.palette.primary.main};
    border-right-width: 1px !important;
    outline: none;
  }
  ${({ $hasError }) =>
    $hasError &&
    css`
      &,
      &:hover,
      &:active,
      &:focus {
        border: 1px solid ${({ theme }) => theme.palette.error.main};
      }
    `}
`
