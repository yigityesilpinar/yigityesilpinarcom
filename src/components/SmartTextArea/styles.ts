import styled from 'styled-components'

import _TextArea from './TextArea'
import { textAreaSpacings } from './TextArea/styles'

export const Container = styled.div``

export const TextLengthCount = styled.span`
  span {
    color: ${({ theme }) => theme.palette.primary.main};
  }
`

export const Formatter = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  ${textAreaSpacings}
  word-break: break-word;
  pointer-events: none;
  /* !important, border mismatch causes text overlap */
  border: 1px solid transparent;
  span {
    white-space: pre-wrap;
  }
  .error {
    color: ${({ theme }) => theme.palette.error.main};
    position: relative;
    border-bottom: 2px dotted ${({ theme }) => theme.palette.error.main};
  }
`

export const TextArea = styled(_TextArea)<{ $smart?: boolean }>`
  textarea {
    color: green;
    -webkit-text-fill-color: transparent;
    ::-webkit-input-placeholder {
      text-shadow: none;
      -webkit-text-fill-color: initial;
    }
  }
`

export const ErrorMessage = styled.div`
  color: ${({ theme }) => theme.palette.error.main};
`
