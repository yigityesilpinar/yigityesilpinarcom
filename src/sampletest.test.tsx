import React from 'react'
import styled from 'styled-components'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { TypographyContainer } from 'src/components/Typography/styles'
import theme from 'src/styles/theme'
import { act } from 'react-dom/test-utils'

const Button = styled.button.attrs(() => ({
  'data-testid': 'my-button-to-test'
}))`
  color: red;
`
interface useCounterOptions {
  initialValue?: number
}
const useCounter = (options?: useCounterOptions) => {
  const { initialValue = 0 } = options || {}
  const [count, setCount] = React.useState(initialValue)
  const increment = () => setCount((c) => c + 1)
  const decrement = () => setCount((c) => c - 1)
  return {
    count,
    increment,
    decrement
  }
}

const CustomHookRender: React.FC<{
  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
  children: (a: any) => null
}> = ({ children, ...restProps }) => children(useCounter(restProps))

const setupHook = (options?: useCounterOptions) => {
  const returnVal = {}
  render(
    <CustomHookRender {...options}>
      {(val) => {
        Object.assign(returnVal, val)
        return null
      }}
    </CustomHookRender>
  )
  return returnVal as ReturnType<typeof useCounter>
}

// Notes:
// use test.only() for isolated test

const asyncExampleFn = ({ fail = false, delay = 2000 }) =>
  new Promise((resolve, reject) => {
    setTimeout(() => (fail ? reject('some error message') : resolve('success')), delay)
  })

describe('Test patterns/examples', () => {
  test('simple test, jest-extended', () => {
    expect(2 + 2).toBe(4)
    // jest-extended
    expect('').toBeString()
  })

  test('styled snapshot test with @testing-library/react', () => {
    expect(render(<Button />).asFragment()).toMatchSnapshot()
  })

  test('using getByTestId', () => {
    const { getByTestId } = render(
      <div>
        hello world
        <Button />
      </div>
    )
    expect(getByTestId('my-button-to-test')).toHaveStyleRule('color', 'red')
  })

  test('using getByTestId again', () => {
    const { getByTestId } = render(
      <div>
        hello world
        <Button />
      </div>
    )
    expect(getByTestId('my-button-to-test')).toHaveStyleRule('color', 'red')
  })

  test('styled snapshot with react-test-renderer and style rule test', () => {
    const tree = renderer.create(<Button />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('color', 'red')
  })

  test('styled render with theme', () => {
    renderer.create(<TypographyContainer theme={theme} />).toJSON()
    expect.assertions(0)
  })

  test('simple async test', async () => {
    const result = await asyncExampleFn({})
    expect(result).toBe('success')
    try {
      await asyncExampleFn({ fail: true })
    } catch (e) {
      expect(e).toMatch(/error/i)
    }
  })

  test('simple hook test', () => {
    const hookResult = setupHook()
    act(() => {
      hookResult.increment()
      hookResult.increment()
      hookResult.increment()
    })

    expect(hookResult.count).toBe(3)
  })

  test('simple hook test with initial value', () => {
    const hookResult = setupHook({ initialValue: 5 })
    act(() => {
      hookResult.increment()
      hookResult.increment()
      hookResult.increment()
    })

    expect(hookResult.count).toBe(8)
  })
})
