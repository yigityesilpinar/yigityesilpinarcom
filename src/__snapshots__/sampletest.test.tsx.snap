// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test patterns/examples styled snapshot test with @testing-library/react 1`] = `
<DocumentFragment>
  .c0 {
  color: red;
}

<button
    class="c0"
    data-testid="my-button-to-test"
  />
</DocumentFragment>
`;

exports[`Test patterns/examples styled snapshot with react-test-renderer and style rule test 1`] = `
.c0 {
  color: red;
}

<button
  className="c0"
  data-testid="my-button-to-test"
/>
`;
